fees/veda.ts

```diff
@@ -0,0 +1,285 @@
+      const token = getTokens[i]
+
+      // get vaults rate updateed events
+      const lendingPoolContract: Interface = new Interface([
+        BoringVaultAbis.exchangeRateUpdated,
+      ])
+      const events: Array<ExchangeRateUpdatedEvent> = (await options.getLogs({
+        eventAbi: BoringVaultAbis.exchangeRateUpdated,
+        entireLog: true,
+        target: accountant,
+      }))
+      .map(log => {
+        const decodeLog: any = lendingPoolContract.parseLog(log);
+
+        const event: any = {
+          blockNumber: Number(log.blockNumber),
+          oldRate: decodeLog.args[0],
+          newRate: decodeLog.args[1],
+        }
+
+        return event
+      })
+      
+      for (const event of events) {
+        // newRate - oldRate
+        const growthRate = event.newRate > event.oldRate ? Number(event.newRate - event.oldRate) : 0;
+
+        // don't need to make calls if there isn't rate growth
+        if (growthRate > 0) {
+          // get total staked in vault at the given block
+          // it's safe for performance because ExchangeRateUpdated events
+          // occur daily once
+          const totalSupplyAtUpdated = await sdk.api2.abi.call({
+            abi: BoringVaultAbis.totalSupply,
+            target: vault.vault,
+            block: event.blockNumber,
+          })
+          const getRateAtUpdated = await sdk.api2.abi.call({
+            abi: BoringVaultAbis.getRate,
+            target: accountant,
+            block: event.blockNumber,
+          })
+
+          // rate is always greater than or equal 1
+          const totalDeposited = Number(totalSupplyAtUpdated) * Number(getRateAtUpdated) / vaultRateBase
+
+          const performanceFeeRate = vault.performanceFee ? vault.performanceFee : 0;
+          const supplySideYield = totalDeposited * growthRate / vaultRateBase
+          const totalYield = supplySideYield / (1 - performanceFeeRate)
+          const protocolFee = totalYield - supplySideYield
+
+          dailyFees.add(token, totalYield)
+          dailySupplySideRevenue.add(token, supplySideYield)
+          dailyProtocolRevenue.add(token, protocolFee)
+        }
+      }
+
+      // get total asset are deposited in vault
+      const totalSupply = await options.api.call({
+        abi: BoringVaultAbis.totalSupply,
+        target: vault.vault,
+      })
+      const getRate = await options.api.call({
+        abi: BoringVaultAbis.getRate,
+        target: accountant,
+      })
+      const totalDeposited = Number(totalSupply) * Number(getRate) / vaultRateBase
+
+      // default one day
+      let timespan = 24 * 60 * 60
+      if (options.fromApi.timestamp && options.toApi.timestamp) {
+        timespan = options.toApi.timestamp - options.fromApi.timestamp
+      }
+
+      // platform fees changred by Veda per year of total assets in vault
+      const yearInSecs = 365 * 24 * 60 * 60
+      const platformFee = totalDeposited * vault.paltformFee / yearInSecs * timespan
```

Comment:
I will feel safer, also easier to read if it was  `(timespan/yearInSecs)` this way there is no room for confusion